h2o-core/src/main/java/hex/ModelMojoWriter.java

```diff
@@ -241,4 +244,14 @@ private void writeDomains() throws IOException {
       finishWritingTextFile();
     }
   }
+
+  /** Create file that contains model details in JSON format.
+   * This information is pulled from the models schema.
+   */
+  private void writeModelDetails() throws IOException{
+    ModelSchemaV3 modelSchema = (ModelSchemaV3) SchemaServer.schema(3, model).fillFromImpl(model);
```

Comment:
PR-800 alone isn't problematic here: you can always add a tiny piece of logic to run schema registration if it hasn't been done by this point. However the method `SchemaServer.schema()` is deprecated, since it doesn't have well-defined semantic: it assumes there is 1-to-1 correspondence between java classes and their schemas, which is not always the case. Besides, this will break if you use any 99 algo, or if we start to gradually transition to REST v4. It would be much safer to have a dedicated factory method for model->schema conversion.

Then of course, there is the question of using REST layer itself to store model's parameters. Although a nice shortcut, it's too error-prone: if anything within a schema changes, the mojo internal format changes, which may not be reflected in appropriate mojo version increase. Also it would make it much harder to make REST component independent from the core h2o.